tests/unit/s2n_cert_chain_and_key_test.c

```diff
@@ -188,6 +188,32 @@ int main(int argc, char **argv)
         };
     };
 
+    /* s2n_cert_chain_and_key_load_pem */
+    {
+        /* when loading a chain, all certs have a info associated with them and root is self-signed */
+        {
+            DEFER_CLEANUP(struct s2n_cert_chain_and_key *cert_chain = NULL,
+                    s2n_cert_chain_and_key_ptr_free);
+            EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&cert_chain, "ec", "ecdsa",
+                    "p384", "sha256"));
+            struct s2n_cert *leaf = cert_chain->cert_chain->head;
+            EXPECT_EQUAL(leaf->info.self_signed, false);
+            EXPECT_EQUAL(leaf->info.signature_nid, NID_ecdsa_with_SHA256);
+            EXPECT_EQUAL(leaf->info.signature_digest_nid, NID_sha256);
+
+            struct s2n_cert *intermediate = leaf->next;
```

Comment:
nit: I feel like it might be worth checking these in case the cert changes or the wrong one is loaded or something.
```suggestion
            struct s2n_cert *intermediate = leaf->next;
            EXPECT_NOT_NULL(intermediate);
```